server/test/api/functions.js

```diff
@@ -0,0 +1,94 @@
+/* global describe, it, beforeEach */
+
+var chai = require('chai');
+var expect = chai.expect;
+var uuid    = require('node-uuid');
+
+var helpers = require('./helpers');
+helpers.configure(chai);
+
+/**
+* The /function  API endpoint
+*
+* This test suite implements full CRUD on the /function  HTTP API endpoint.
+*/
+describe('The /function  API endpoint', function () {
```

Comment:
All of these routes should be called `/functions`, not `/function`.  You have mislabeled all of them.  Please update the labels for every test so that we are not confused when we come to debug.